presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/PartitionTable.java

```diff
@@ -339,8 +339,8 @@ public Partition(
             this.recordCount = recordCount;
             this.fileCount = 1;
             this.size = size;
-            this.minValues = ImmutableMap.copyOf(requireNonNull(minValues, "minValues is null"));
-            this.maxValues = ImmutableMap.copyOf(requireNonNull(maxValues, "maxValues is null"));
+            this.minValues = new HashMap<>(requireNonNull(minValues, "minValues is null"));
+            this.maxValues = new HashMap<>(requireNonNull(maxValues, "maxValues is null"));
```

Comment:
shouldn't update stats produce new `Partition` object or at least `minValues/maxValues` maps.
Is `Partition` used by multiple threads?